mono/metadata/w32handle.c

```diff
@@ -318,17 +462,23 @@ mono_w32handle_ops_typesize (MonoW32HandleType type);
 static void mono_w32handle_init_handle (MonoW32HandleBase *handle,
 			       MonoW32HandleType type, gpointer handle_specific)
 {
-	g_assert (handle->ref == 0);
+	guint old_state = assert_handle_data_is_available (handle);
```

Comment:
It's weird, but kinda makes sense, reading the same var twice would yield different results.
Would renaming it to something more sensible address this?